servers/rendering/renderer_rd/storage_rd/texture_storage.cpp

```diff
@@ -2480,10 +2480,17 @@ void TextureStorage::_update_render_target(RenderTarget *rt) {
 	if (rt->size.width == 0 || rt->size.height == 0) {
 		return;
 	}
-	//until we implement support for HDR monitors (and render target is attached to screen), this is enough.
-	rt->color_format = RD::DATA_FORMAT_R8G8B8A8_UNORM;
-	rt->color_format_srgb = RD::DATA_FORMAT_R8G8B8A8_SRGB;
-	rt->image_format = rt->is_transparent ? Image::FORMAT_RGBA8 : Image::FORMAT_RGB8;
+	if (rt->viewport_mode == RS::VIEWPORT_MODE_3D) {
+		// For 3D only viewport keep precision
+		rt->color_format = RD::DATA_FORMAT_R16G16B16A16_SFLOAT;
```

Comment:
So here the same thing, we should not hardcode `RD::DATA_FORMAT_R16G16B16A16_SFLOAT` here. The mobile renderer does not use an RGBA16F buffer, those don't work well at all. We use RGB10A2 buffers here with all color values divided by 2. That's going to be interesting for people who don't realize they have to multiply by 2 when accessing it (this is the luminance multiplier you see in various shaders).

You can call `_render_buffers_get_color_format` to get the format you need to use here.
You can call `_render_buffers_get_luminance_multiplier` to get the luminance multiplier if we want to write something that detects we're using a viewport like this.

Reply:
Thanks for the precision, i'm indeed a bit confused by how luminance is applied. I'll try to use the mentionned method to be more reliable.

Reply:
Basically the problem on mobile GPUs is that they suck when you're using a 64bit value to store color data in. 32bit formats are optimised and often have hardware compression to minimize bandwidth while loading/saving data from tile memory.

Hence we use RGB10_A2 but since this does not store a floating point value, but is rather a 10bit uint value that is normalized to `0.0 - 1.0`, we divide the color by 2 to extend this range to '0.0 - 2.0` and have some measure of higher dynamic range. We've still got more precision that when using 8bits so it prevents color banding unless you have very dark scenes that you're making brighter by exposure.

The 2.0 is a fixed ceiling so in very bright scenes you just loose the upper band.

Anyway, that is why we do it, the result is that all colors need to be multiplied by 2 when read from the texture to regain their original value.

Reply:
Note also that if we set transparent on, we do default back to `RGBA8` as we're assuming tonemapping and sRGB conversion has been applied by now seeing `RGB10_A2` only has 2 bits of transparency.

Using `RGB10_A2` here really only makes sense if we're not applying post processing. That is the only scenario where using `RGB8` would cause banding as it lacks the precision to store linear color data.